sdk/src/azure/core/az_http_request.c

```diff
@@ -107,11 +171,45 @@ az_http_request_set_query_parameter(_az_http_request* ref_request, az_span name,
   // name or value can't be empty
   _az_PRECONDITION(az_span_size(name) > 0 && az_span_size(value) > 0);
 
-  int32_t required_length = az_span_size(name) + az_span_size(value) + 2;
-
+  // check if query parameter is already in url
+  az_span prexisting_query_parameter_value
+      = _az_http_request_find_query_parameter(ref_request, name);
   az_span url_remainder
       = az_span_slice_to_end(ref_request->_internal.url, ref_request->_internal.url_length);
 
+  if (az_span_ptr(prexisting_query_parameter_value) != NULL)
+  { // a negative difference means shifting left and no required length.
+    // a positive difference means new value will require shifting right.
+    int32_t prexisting_query_parameter_value_size = az_span_size(prexisting_query_parameter_value);
+    int32_t difference = az_span_size(value) - prexisting_query_parameter_value_size;
+    int32_t required_length = difference > 0 ? difference : 0;
+    AZ_RETURN_IF_NOT_ENOUGH_SIZE(url_remainder, required_length);
+
+    // There is already a query parameter name. update the value if different
+    if (az_span_is_content_equal(value, prexisting_query_parameter_value))
+    {
+      // No need to do anything, existing value is equal to the new one
+      return AZ_OK;
+    }
+
+    uint8_t* url_prt_start = az_span_ptr(ref_request->_internal.url);
+    // ptr difference should work because value is always inside url, same memory
+    int32_t value_index = (int32_t)(az_span_ptr(prexisting_query_parameter_value) - url_prt_start);
```

Comment:
nit: use the existing `_az_span_diff(...)` helper method to reduce duplication/calls to `az_span_ptr`).

Reply:
I was forgetting about that API!  thank you!

Reply:
just realized this API does not work here.
The API returns the diff after slicing one az_span B out of another az_span A, by doing sizeA - sizeB

What I need is just the distance from the start A to the start B. It works if B ends at same position as B, but if B ends before, this distance is not the one I want because I would get  also any distance after B

Reply:
Yes, you are right. That method returns the difference in size, not the ptr diff. I was misremembering and the overlaps helper just returns a bool, not the integer difference.